app/server/lib/ActiveDoc.ts

```diff
@@ -3182,23 +3189,28 @@ export class ActiveDoc extends EventEmitter {
     // Minor flaw: while we don't double-count existing duplicate files in the total size,
     // we don't check here if any of the uploaded files already exist and could be left out of the calculation.
     const uploadSizeBytes = sum(upload.files.map(f => f.size));
-    if (await this._isUploadSizeBelowLimit(uploadSizeBytes)) { return; }
-
-    // TODO probably want a nicer error message here.
-    throw new LimitExceededError("Exceeded attachments limit for document");
-  }
-
-  /**
-   * Returns true if an upload with size `uploadSizeBytes` won't cause attachment size
-   * limits to be exceeded.
-   */
-  private async _isUploadSizeBelowLimit(uploadSizeBytes: number): Promise<boolean> {
-    const maxSize = this._product?.features.baseMaxAttachmentsBytesPerDocument;
-    if (!maxSize) { return true; }
 
     let currentSize = this._docUsage?.attachmentsSizeBytes;
     currentSize = currentSize ?? await this._updateAttachmentsSize({syncUsageToDatabase: false});
-    return currentSize + uploadSizeBytes <= maxSize;
+    const futureSize = currentSize + uploadSizeBytes;
+
+    const productMaxSize = this._product?.features.baseMaxAttachmentsBytesPerDocument;
+
+    if (productMaxSize !== undefined && futureSize > productMaxSize) {
+      // TODO probably want a nicer error message here.
+      throw new LimitExceededError("Exceeded attachments limit for document");
+    }
+
+    // If not using external attachments, apply installation limit on size.
+    // Technically the attachmentStoreId being set doesn't mean all attachments
+    // are actually stored externally, and a determined person could
+    // work around this limit at the API level. That's fine, their
+    // reward will be pain.
```

Comment:
Agreed! I can imagine an in-UI workaround for this too (set to external, upload, transfer to internal).

I think either:
- This is completely fine, and pain shall become their companion
- We apply the limit during transfers too

I'll leave that up to you!

Reply:
I added a thing where if starting a transfer to internal, limits are checked. Since it is easy to imagine a doc with a lot of attachments stored externally, then user clicking to make internal to e.g. make the doc standalone. Let me know if it looks ok. Didn't get into nuance of trying to free up space if there are lingering attachments on the road to deletion. It is a little silly since all transfers will fail if total attachment size is greater than limit, and communication with user is poor about errors. But it could be easily improved if this is a corner users do in practice find themselves in.